samples/browser/src/main/java/org/mozilla/samples/browser/addons/AddonDetailsActivity.kt

```diff
@@ -77,6 +88,31 @@ class AddonDetailsActivity : AppCompatActivity() {
     private fun bindVersion(addon: Addon) {
         val versionView = findViewById<TextView>(R.id.version_text)
         versionView.text = addon.installedState?.version?.ifEmpty { addon.version } ?: addon.version
+
+        if (addon.isInstalled()) {
+            versionView.setOnLongClickListener {
+                showUpdaterDialog(addon)
+                true
+            }
+        }
+    }
+
+    private fun showUpdaterDialog(addon: Addon) {
+        val context = this
+        val scope = CoroutineScope(IO)
+        val storage = DefaultAddonUpdater.RequestStorage(this)
+
+        scope.launch {
+            val request = storage.findRequestBy(addon.id)
```

Comment:
Let's make this re-usable by either adding a suspend method to `AddonUpdater` or `AddonManager`. Something like `getAddonUpdateStatus(context: Context)` which returns a String? This way we can limit duplication in our apps. We can then also move `getDialogMessage` away from the app.

Reply:
Definitely important to lazy initialize the storage / room db, but we can hide that from the app by doing it internally once needed.

Reply:
I extracted the logic for getting the string and showing the dialog. The only left for the apps to do, it is to fetch the information using the storage.

Reply:
Let me know what do you think!